packages/gator-permissions-snap/test/core/permissionRequestLifecycleOrchestrator.test.ts

```diff
@@ -560,12 +571,36 @@ describe('PermissionRequestLifecycleOrchestrator', () => {
           'Expiry rule not found. An expiry is required on all permissions.',
         );
       });
+      it('checks account upgrade status and triggers upgrade when needed', async () => {
```

Comment:
Consider to add test for already-upgraded accounts, for the already-upgraded accounts, the `AccountController.upgradeAccount` should be expected not to be called.